pkg/destinations/bigquery/query.go

```diff
@@ -0,0 +1,129 @@
+package bigquery
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/api/iterator"
+)
+
+func (b *BigQueryServer) QueryJSON(query string, writer io.Writer) error {
+	// NOTE: Query should be with dataset as prefix. Example: SELECT * FROM `dataset.table`
+
+	ctx := context.TODO()
+	q := b.conn.Query(query)
+
+	writer.Write([]byte("["))
+	enc := json.NewEncoder(writer)
+
+	itr, err := q.Read(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("error getting query iterator")
+		return err
+	}
+	firstRow := true
+	for {
+		var data map[string]bigquery.Value
+		err := itr.Next(&data)
+
+		if err == iterator.Done {
+			break
+		}
+		if !firstRow {
+			_, err = writer.Write([]byte(","))
+			if err != nil {
+				log.Error().Err(err).Msg("failed to write JSON array separator")
+				return err
+			}
+		} else {
+			firstRow = false
+		}
+
+		err = enc.Encode(data)
+		if err != nil {
+			return err
+		}
+	}
+	writer.Write([]byte("]"))
+	return nil
+}
+
+func (b *BigQueryServer) QueryCSV(query string, writer io.Writer) error {
+	// NOTE: Query should be with dataset as prefix. Example: SELECT * FROM `dataset.table`
+
+	ctx := context.TODO()
+	q := b.conn.Query(query)
+
+	enc := csv.NewWriter(writer)
+
+	columnsQuery := b.conn.Query(query)
+	columnsItr, err := columnsQuery.Read(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("error getting columns query iterator")
+		return err
+	}
+
+	columns := make([]string, 0)
+	columnsRow := make(map[string]bigquery.Value)
+	if err := columnsItr.Next(&columnsRow); err != nil {
+		log.Error().Err(err).Msg("error retrieving columns")
+		return err
+	}
+
+	//column orders are not ensured in bigquery
+	columns = append(columns, "__row_id")
```

Comment:
We don't care about making the __row_id the first column.

Reply:
Okay then removing!